integration/utils/tx_result.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sethvargo/go-retry"
+
+	flowsdk "github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/access"
+)
+
+// GetTransactionResult waits for the transaction to get into the terminal state and returns the result.
+func GetTransactionResult(ctx context.Context, client access.Client, txID flowsdk.Identifier) *flowsdk.TransactionResult {
+	var b retry.Backoff
+	b = retry.NewFibonacci(100 * time.Millisecond)
+	b = retry.WithMaxDuration(60*time.Second, b)
+	b = retry.WithCappedDuration(10*time.Second, b)
+
+	var result *flowsdk.TransactionResult
+	err := retry.Do(ctx, b, func(ctx context.Context) (err error) {
+		result, err = client.GetTransactionResult(ctx, txID)
+		if err != nil {
+			return err
+		}
+		if result.Error != nil {
+			return result.Error
+		}
+
+		switch result.Status {
+		case flowsdk.TransactionStatusExecuted, flowsdk.TransactionStatusSealed:
+			return nil
+		case flowsdk.TransactionStatusPending, flowsdk.TransactionStatusFinalized:
+			return retry.RetryableError(fmt.Errorf("waiting for transaction execution: %s", txID))
+		default:
+			return fmt.Errorf("unexpected transaction status: %s", result.Status)
+		}
+	})
+	if err != nil {
+		return &flowsdk.TransactionResult{Error: err}
```

Comment:
just return `(*TransactionResult, error)` tuple instead of wrapping the error?

Reply:
Sadly the `TransactionResult` does have an error field, so downstream code usually checks it anyway.  That said, let me try duplicating the error in return statement -- this would likely be cleaner.